src/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs

```diff
@@ -1171,6 +1171,34 @@ public void BufferList_InvalidArguments_Throws(int length, int offset, int count
                 Assert.Throws<ArgumentOutOfRangeException>(() => e.BufferList = new List<ArraySegment<byte>> { validBuffer, invalidBuffer });
             }
         }
+
+        [Fact]
+        public async Task Recv_PeerDisconnect_Returns()
```

Comment:
Nit: could you move this to next to the `SendRecv_0ByteReceive_Success` and also similarly include something about 0 bytes in the name?